tests/integration/backup_tests/test_backups.py

```diff
@@ -71,20 +72,31 @@ async def add_writes_to_db(ops_test: OpsTest):
     await clear_writes_action.wait()
 
 
+@pytest.mark.skipif(
+    os.environ.get("PYTEST_SKIP_DEPLOY", False),
+    reason="skipping deploy, model expected to be provided.",
+)
 @pytest.mark.abort_on_fail
 async def test_build_and_deploy(ops_test: OpsTest) -> None:
     """Build and deploy one unit of MongoDB."""
     # it is possible for users to provide their own cluster for testing. Hence check if there
     # is a pre-existing cluster.
-    db_app_name = await ha_helpers.get_application_name(ops_test, DATABASE_APP_NAME)
+    db_app_name = await get_app_name(ops_test)
+    num_units = 3
     if db_app_name:
-        return
-
-    async with ops_test.fast_forward():
-        my_charm = await ops_test.build_charm(".")
-        resources = {"mongodb-image": METADATA["resources"]["mongodb-image"]["upstream-source"]}
-        await ops_test.model.deploy(my_charm, num_units=3, resources=resources, series="jammy")
-        await ops_test.model.wait_for_idle(apps=[DATABASE_APP_NAME], status="active", timeout=2000)
+        await check_or_scale_app(ops_test, db_app_name, num_units)
```

Comment:
does `check_or_scale_app` - wait for all units to be idle after scaling? If so, resolve this comment, otherwise, make sure that the deploy_test waits for this

Reply:
done: updated `check_or_scale_app` to make it wait for idle